### ISSUE TITLE
DateOffset & Timezones

### ISSUE THREAD

**Issue by jonathanng** [pydata/pandas#9820]
tz = pytz.timezone('America/New_York')
    tz.localize(datetime.datetime(2007,3,12)) - DateOffset(days=1)
    # Timestamp('2007-03-11 00:00:00-0400', tz='America/New_York')

Shouldn't this be?
    
    Timestamp('2007-03-11 00:00:00-0500', tz='America/New_York')

**Comment by rockg** [comment]
It works on the current master.  There was an issue with some offset classes not doing the right thing with timezones.

```
>>> tz.localize(datetime(2007, 3, 12)) - pd.DateOffset(days=1)
Timestamp('2007-03-11 00:00:00-0500', tz='US/Eastern')
```

**Comment by jorisvandenbossche** [comment]
@jonathanng Thanks for reporting anyhow! (and @rockg thanks for checking)

Note this not only works in master, but also in the recently released 0.16.0
